indico/modules/events/papers/models/papers.py

```diff
@@ -66,3 +68,45 @@ def state(self):
     @state.setter
     def state(self, state):
         self.last_revision.state = state
+
+    def is_in_final_state(self):
+        return self.state in {PaperRevisionState.accepted or PaperRevisionState.rejected}
+
+    def can_comment(self, user):
+        if not user:
+            return False
+        return self.is_submitter(user) or self.can_judge(user) or self.can_review(user)
+
+    def is_submitter(self, user):
+        if not user:
+            return False
+        return user in {r.submitter for r in self.revisions} or user in self.contribution.submitters
+
+    def can_manage(self, user):
+        if not user:
+            return False
+        return self.contribution.can_manage(user)
+
+    def can_judge(self, user, check_state=False):
+        if not user:
+            return False
+        elif check_state and self.is_in_final_state():
+            return False
+        elif self.can_manage(user):
+            return True
+        return user in self.contribution.paper_judges
+
+    def can_review(self, user, check_state=False):
+        if not user:
+            return False
+        elif check_state and self.is_in_final_state():
+            return False
+        elif self.can_manage(user):
+            return True
+        return user in self.contribution.paper_content_reviewers or user in self.contribution.paper_layout_reviewers
+
+    def get_revisions(self):
```

Comment:
Rly? It raises an `NotImplementedError`, why document it? Isn't it enough? Then we would have to do the same to all methods that raise such an error!